Automattic/newspack-custom-content-migrator
# Add a "block-transformer" that "hides" blocks for the NCC

This is a bit of an experimental attempt at getting mixed classic and GB blocks to survive an NCC run.

The idea is that we base64-encode all GB blocks before running NCC and then decode them after NCC is done.

It's for specific use for a publisher now and certainly needs some docs and polishing, but it should be kinda resuable. I'm also not sure I love the name of the class, so we can change that if we find a better name.

## How to test
- To encode, run `wp newspack-content-migrator transform-blocks-encode --post-id=XXX`
- Run NCC on the post
- Decode with `wp newspack-content-migrator transform-blocks-decode --post-id=XXX`

The post should have survived without being chopped to pieces by NCC or by bad decoding.

---
